test/mimc.js

```diff
@@ -0,0 +1,96 @@
+const mimcjs = require("../circomlib/src/mimc7.js");
+const mimcGenContract = require("../circomlib/src/mimc_gencontract.js");
+
+/*
+    Here we want to test the hashing of the MIMC7 contract
+*/ 
+
+contract("MiMC7 Hashing", async accounts => {
+    let mimc;
+    const SEED = "mimc";
+
+    it("Should deploy the contract", async () => {
+        const C = new web3.eth.Contract(mimcGenContract.abi);
+
+        mimc = await C.deploy({
+            data: mimcGenContract.createCode(SEED, 91)
+        }).send({
+            gas: 1500000,
+            from: accounts[0]
+        });
+    });
+
+    it("Should return a hash", async () => {
+        const input = [2];
+        const key = 22;
+
+        let result = await mimc.methods.MiMCpe7(input, key).call();
+        // console.log(result);
+        // console.log(result.toString());
+    });
+
+    it("Should hash value correctly in js", async () => {
+        const m = BigInt(3703141493535563179657531719960160174296085208671919316200479060314459804651).toString();
+        const k = BigInt(918403109389145570117360101535982733651217667914747213867238065296420114726).toString();
+        
+        const jsExpected = mimcjs.hash(m, k);
+        const solidityResult = await mimc.methods.MiMCpe7(m, k).call();
+
+        assert.equal(solidityResult.toString(), jsExpected.toString(), "Unexpected result");
+    });
+
+    it("Should hash value correctly in solidity", async () => {
+        const res = await mimc.methods.MiMCpe7(1,2).call();
+        const res2 = await mimcjs.hash(1,2,91);
+
+        assert.equal(res.toString(), res2.toString());
+    });
+
+    
+    /*
+
+              G
+            E    F
+          A  B  C  D
+
+        E = hash(A, B)
+        F = hash(C, D)
+        G = hash(E, F)
+
+        multihash(a,b,c,d) == G
+    */
+    it("Should parse trees the same way as the multihash", async () => {
+        const a = BigInt(123).toString()
+        const b = BigInt(564).toString()
+        const c = BigInt(894).toString()
+        const d = BigInt(354).toString()
+
+        const solE = await mimc.methods.MiMCpe7(a,b).call();
+        const jsE = mimcjs.hash(a,b,91);
+        assert.equal(solE.toString(), jsE.toString());
+        
+        const solF = await mimc.methods.MiMCpe7(c,d).call();
+        const jsF = mimcjs.hash(c,d,91);
+        assert.equal(solF.toString(), jsF.toString());
+        
+        const solG = await mimc.methods.MiMCpe7(solE,solF).call();
+        const jsG = mimcjs.hash(jsE,jsF,91);
+        assert.equal(solG.toString(), jsG.toString());
+
+        // TODO: parse it with ying tong's MiMCMerkle.js after merge
```

Comment:
@therealyingtong Do you think we'll be able to use the `MiMCMerkle` contract from ethsnarks? We could simply extend the contract and provide our own version of `HashImpl` to make it work with `mimc7.js` from circomlib.

See: https://github.com/HarryR/ethsnarks/blob/master/contracts/MerkleTree.sol#L58

Reply:
If we do end up using ethsnarks MiMC implementation, could be nice to also use their MiMCMerkle.